esx_service/utils/kvESX.py

```diff
@@ -42,130 +47,156 @@
 
 # Open a VMDK given its path, the VMDK is opened locked just to
 # ensure we have exclusive access and its not already in use.
 def volOpenPath(volpath):
-    dhandle = c_uint32(0)
-    ihandle = c_uint32(0)
+    dhandle = get_uint(0)
+    ihandle = get_uint(0)
     key = c_uint32(0)
 
-    res = lib.DiskLib_OpenWithInfo(volpath, VMDK_OPEN_FLAGS, byref(key),
+    res = lib.DiskLib_OpenWithInfo(volpath.encode(), VMDK_OPEN_FLAGS, byref(key),
                                    byref(dhandle), byref(ihandle))
 
     if res != 0:
         logging.warning("Open %s failed - %x", volpath, res)
 
     return dhandle
 
-
 # Create the side car for the volume identified by volpath.
 def create(volpath, kv_dict):
-    disk = c_uint32(0)
-    objHandle = c_uint32(0)
-
-    disk = volOpenPath(volpath)
-
-    if disk == c_uint32(0):
-        return False
-
-    if useSideCarCreate:
-        res = lib.DiskLib_SidecarCreate(disk, DVOL_KEY, KV_CREATE_SIZE,
-                                        KV_SIDECAR_CREATE, byref(objHandle))
-    else:
-        res = lib.DiskLib_SidecarOpen(disk, DVOL_KEY, KV_SIDECAR_CREATE,
-                                      byref(objHandle))
+    objHandle = get_uint(0)
+    dhandle = volOpenPath(volpath)
 
-    if res != 0:
-        logging.warning("Side car create for %s failed - %x", volpath, res)
-        lib.DiskLib_Close(disk)
-        return False
+    if disk_is_valid(dhandle):
+       if useSideCarCreate:
+          res = lib.DiskLib_SidecarCreate(dhandle, DVOL_KEY.encode(), KV_CREATE_SIZE,
+                                          KV_SIDECAR_CREATE, byref(objHandle))
+       else:
+          res = lib.DiskLib_SidecarOpen(dhandle, DVOL_KEY.encode(), KV_SIDECAR_CREATE,
+                                        byref(objHandle))
+       if res != 0:
+          logging.warning("Side car create for %s failed - %x", volpath, res)
+          lib.DiskLib_Close(dhandle)
+          return False
 
-    lib.DiskLib_SidecarClose(disk, DVOL_KEY, byref(objHandle))
-    lib.DiskLib_Close(disk)
+       lib.DiskLib_SidecarClose(dhandle, DVOL_KEY.encode(), byref(objHandle))
+       lib.DiskLib_Close(dhandle)
 
-    return save(volpath, kv_dict)
+       return save(volpath, kv_dict)
 
 
 # Delete the the side car for the given volume
 def delete(volpath):
-    disk = c_uint32(0)
+    dhandle = volOpenPath(volpath)
 
-    disk = volOpenPath(volpath)
-
-    if disk == c_uint32(0):
-        return False
+    if disk_is_valid(dhandle):
+       res = lib.DiskLib_SidecarDelete(dhandle, DVOL_KEY.encode())
+       if res != 0:
+          logging.warning("Side car delete for %s failed - %x", volpath, res)
+          lib.DiskLib_Close(dhandle)
+          return False
 
-    res = lib.DiskLib_SidecarDelete(disk, DVOL_KEY)
-    if res != 0:
-        logging.warning("Side car delete for %s failed - %x", volpath, res)
-        lib.DiskLib_Close(disk)
-        return False
-
-    lib.DiskLib_Close(disk)
-    return True
+       lib.DiskLib_Close(dhandle)
```

Comment:
Same issue as my comment on create. In this function however, it appears that returning None is absolutely incorrect and False should be returned if `disk_is_valid` returns false.